src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,7 +208,14 @@ private async Task LoadAll(bool ignoreFailures)
                     {
                         await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsAsync(selector, CancellationToken.None).ConfigureAwait(false))
                         {
-                            serverData[setting.Key] = setting;
+                            if (serverData.ContainsKey(setting.Key))

```

Comment:
>if (serverData.ContainsKey(setting.Key)) [](start = 28, length = 40)

When would this condition be true in this code path?

Reply:
Good point, since we fetch everything with null label first, this condition would never be true. It would always go to the else block. I'll remove this check.